lib/puppet/pops/issues.rb

```diff
@@ -387,6 +387,12 @@ def self.hard_issue(issue_code, *args, &block)
     _("Expressions of type %{expression} are not supported in this version of Puppet") % { expression: label.a_an(semantic) }
   end
 
+  # Issues when expressions that are not implemented or activated when running with --tasks
+  #
+  UNSUPPORTED_TASKS_EXPRESSION = issue :UNSUPPORTED_TASKS_EXPRESSION do
```

Comment:
I added two errors already for similar things and I think that is already perhaps one too many for the same thing. Possibly use CATALOG_OPERATION_NOT_SUPPORTED_WHEN_SCRIPTING. The phrasing there was choosen because users are not really aware that there is a --tasks options when using the "other to be released CLI" so I phrased it "is only available when compiling a catalog".